components/builder-depot/src/server.rs

```diff
@@ -1764,26 +1764,7 @@ fn show_package(req: &mut Request) -> IronResult<Response> {
         request.set_ident(ident.clone());
 
         match route_message::<OriginPackageGet, OriginPackage>(req, &request) {
-            Ok(pkg) => {
-                let lock = req.get::<persistent::State<Config>>()
-                    .expect("depot not found");
-
-                let depot = lock.read().expect("depot read lock is poisoned");
-                let pkg_target = PackageTarget::from_str(&target);
-
-                // If the target doesn't parse or there's no archive on disk, return 404
-                if pkg_target.is_err() || !depot.archive(&ident, &pkg_target.unwrap()).is_some() {
-                    return Ok(Response::with(status::NotFound));
```

Comment:
This error handling for the target feels valuable to keep, but is probably best moved to the top of this function where we create the `target` variable.  `target_from_headers` already makes the `PackageTarget::from_str()` call, but we blindly unwrap it instead of checking for errors and returning `NotFound`.

Also, we should be checking the target that we get from the URL, if one exists.

Reply:
Good point - yes, a target check up front and removal / removal of blind unwraps seems like the right thing - I'll play with it a bit more.